src/model/CoilUserDefined.cpp

```diff
@@ -0,0 +1,589 @@
+      bool result = setPointer(OS_Coil_UserDefinedFields::AirOutletTemperatureActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setAirOutletHumidityRatioActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::AirOutletHumidityRatioActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setAirMassFlowRateActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::AirMassFlowRateActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setPlantMinimumMassFlowRateActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::PlantMinimumMassFlowRateActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setPlantMaximumMassFlowRateActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::PlantMaximumMassFlowRateActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setPlantDesignVolumeFlowRateActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::PlantDesignVolumeFlowRateActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setPlantMassFlowRateActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::PlantMassFlowRateActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+    bool CoilUserDefined_Impl::setPlantOutletTemperatureActuator(const EnergyManagementSystemActuator& energyManagementSystemActuator) {
+      bool result = setPointer(OS_Coil_UserDefinedFields::PlantOutletTemperatureActuator, energyManagementSystemActuator.handle());
+      return result;
+    }
+
+  }  // namespace detail
+
+  CoilUserDefined::CoilUserDefined(const Model& model) : WaterToAirComponent(CoilUserDefined::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::CoilUserDefined_Impl>());
+
+    bool ok = true;
+    // setup required Actuators
+    EnergyManagementSystemActuator aota(this->cast<ModelObject>(), "Air Connection 1", "Outlet Temperature");
+    aota.setName("airOutletTemperature");
+    ok = setAirOutletTemperatureActuator(aota);
+    OS_ASSERT(ok);
+    EnergyManagementSystemActuator aohra(this->cast<ModelObject>(), "Air Connection 1", "Outlet Humidity Ratio");
+    aohra.setName("airOutletHumidityRatio");
+    ok = setAirOutletHumidityRatioActuator(aohra);
+    OS_ASSERT(ok);
```

Comment:
Ok, now I get why the IDD has these fields added, and how a single Ctor that takes in no parameter makes sense.